### DISCUSSION TITLE
If the community is solving for plantuml integration - how?

### DISCUSSION THREAD

**Discussion by mdeliw** [squidfunk/mkdocs-material#2389] [General]
Hello, 

Just looking for approaches tried. 

- https://github.com/amarraja/mkdocs-material-plantuml (not working for me yet)
- https://github.com/mikitex70/plantuml-markdown (not working for me yet)
- https://github.com/christo-ph/mkdocs_build_plantuml (worked - but doesn't support inline)

Thanks.

**Comment by facelessuser** [comment]
There's not much to solve. It's just a matter of someone packaging something up. I think with SuperFences you could easily give it a custom formatter for something like a `plantuml` code type, and simply call plantuml and inject the returned image.

This is 100% doable. As the author of SuperFences,  I think the framework is there to allow this, but one thing I cannot do is implement and maintain all the various ways in which people want to leverage SuperFences. It's why I have docs describing how to use Mermaid, but I don't offer a package to auto inject these things into mkdocs and such.

If someone is interested in writing one and has questions, I can certainly answer them.

**Comment by loganknecht** [comment]
Oooooof.

I ended up here after asking the `mkdocs` team if they had a suggestion for a default docker image that includes `plantuml` integration. They stated it's not really their wheel house since they're a tool, and the image I am using is `squidfunk/mkdocs-material`.
See: https://github.com/mkdocs/mkdocs/issues/2465

I've managed to find a decent enough solution for my organization, but does anyone else have a suggestion for a default docker image?

**Comment by mschoettle** [comment]
It works with `plantuml-markdown` and adding it to the `markdown_extensions`:

```yml
markdown_extensions:
    - plantuml_markdown:
          server: http://www.plantuml.com/plantuml
```
